### ISSUE TITLE
🐛 Bug Report(kubernetes-plugin):  Unable to fetch native k8s resources which doesn't have group

### ISSUE THREAD

**Issue by invincibleJai** [backstage/backstage#17865]
### 📜 Description

Unable to fetch native k8s resources which don't have a group like ReplicationController 

```
  customResources:
    - group: ''
      apiVersion: 'v1'
      plural: 'replicationcontrollers'
```

ReplicationController is not in the default object types https://backstage.io/docs/features/kubernetes/configuration/#objecttypes-optional and in customResources group is required field so can't pass empty strip or skip it.

### 👍 Expected behavior

Should be able to fetch native k8s resources which don't have group

### 👎 Actual Behavior with Screenshots

on `yarn dev` it shows error

<img width="890" alt="image" src="https://github.com/backstage/backstage/assets/5129024/8bed5df9-2f5c-4ff9-8be1-1fa408b1c1cc">

### 👟 Reproduction steps

- Setup [backstage k8s plugin](https://backstage.io/docs/features/kubernetes/) (install and configure)
- in `app-config.yaml` provide `ReplicationController` in customResources as below

```
  customResources:
    - group: ''
      apiVersion: 'v1'
      plural: 'replicationcontrollers'
```
- yarn dev

### 📃 Provide the context for the Bug.

I want to show additional resources on Kubernetes page.

### 🖥️ Your Environment

- NodeJS Version (v14): v16
- Operating System and Version (e.g. Ubuntu 14.04): macos Ventura

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by jamieklassen** [comment]
Maybe this makes more sense as a feature request for `objectTypes` to allow specifying additional types beyond the default? In #15761 this note in the [docs](https://backstage.io/docs/features/kubernetes/configuration/#objecttypes-optional) was added:

> it's currently not supported to fetch object types other than the ones specified in the default types.

So we've got some evidence that @namco1992 at least was thinking about this capability too. @mclarke47 wdyt?
